app/static/js/darkmode.js

```diff
@@ -0,0 +1,28 @@
+const ROOT = document.documentElement;
+window.addEventListener("DOMContentLoaded", (event) => {
+  const button = document.querySelector("#darkmode");
+  let isDarkMode = localStorage.getItem("isDarkMode") == "true";
+  setThemeMode(isDarkMode, button, ROOT);
+  button.addEventListener("click", (event) => {
+    isDarkMode = !isDarkMode;
+    localStorage.setItem("isDarkMode", isDarkMode);
+    setThemeMode(isDarkMode, button, ROOT);
+  });
+});
+
+function changeIcon(mode) {
+  const modeButton = document.querySelector("#darkmode");
+  modeButton.name = mode;
+}
+
+function setThemeMode(isDarkMode, button, root) {
+  if (isDarkMode) {
+    root.setAttribute("color-mode", "dark");
```

Comment:
You can use `.dataset` instead

Reply:
as i read setAttribute is faster...

Reply:
That was true few years ago. Now it seems that [datasets] have even better support than [setAttribute].

[datasets]: https://caniuse.com/dataset
[setAttribute]: https://caniuse.com/?search=setAttribute

Reply:
Talking about the performance, it's [almost neglectable](https://www.measurethat.net/Benchmarks/Show/9389/0/modern-dataset-vs-old-getattribute-vs-jquery-data-vs-jq#latest_results_block).